src/components/Table/TableHead/TableHead.jsx

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { DataTable, Checkbox } from 'carbon-components-react';
+import isNil from 'lodash/isNil';
+
+import ColumnHeaderRow from '../ColumnHeaderRow/ColumnHeaderRow';
+import FilterHeaderRow from '../FilterHeaderRow/FilterHeaderRow';
+
+const { TableHead: CarbonTableHead, TableRow, TableExpandHeader, TableHeader } = DataTable;
+
+const propTypes = {
+  /** Important table options that the head needs to know about */
+  options: PropTypes.shape({ hasRowExpansion: PropTypes.bool, hasRowSelection: PropTypes.bool }),
+  /** List of columns */
+  columns: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      name: PropTypes.string.isRequired,
+      size: PropTypes.number.isRequired,
+      isSortable: PropTypes.bool,
+      filter: PropTypes.shape({
+        placeholderText: PropTypes.string,
+        options: PropTypes.arrayOf(
+          PropTypes.shape({
+            id: PropTypes.string.isRequired,
+            text: PropTypes.string.isRequired,
+          })
+        ),
+      }),
+    })
+  ).isRequired,
+
+  /** Current state of the table */
+  state: PropTypes.shape({
```

Comment:
can we call this something different, to avoid overloading the "state" word?  `tableState` would suffice IMO

Reply:
or `tableView`, since that is the prop we are deconstructing to pass through here

Reply:
good idea